reView/utils/plots.py

```diff
@@ -0,0 +1,284 @@
+# -*- coding: utf-8 -*-
+"""Methods for creating static, report quality graphics.
+
+@author: Mike Gleason
+"""
+import warnings
+
+import numpy as np
+import mapclassify as mc
+from matplotlib.patheffects import SimpleLineShadow, Normal
+import geoplot as gplt
+
+
+# pylint: disable=no-member, arguments-differ
+class YBFixedBounds(np.ndarray):
+    """
+    Helper class for use with a mapclassify classifer. This can used to
+    overwrite the ``yb`` property of a classifier so that the .max() and .min()
+    methods return preset values rather than the maximum and minimum break
+    labels corresponding to the range of data.
+
+    This is used in map_supply_curve_column() to ensure that breaks and colors
+    shown in the legend are always consistent with the input ``breaks`` rather
+    than subject to change based on range of the input ``column``.
+    """
+    def __new__(cls, input_array, preset_max, preset_min=0):
+        """
+        Return a new instance of the class with fixed maximum and minimum
+        values.
+
+        Parameters
+        ----------
+        input_array : numpy.ndarray
+            Input numpy array, typically sourced from the ``yb`` property of a
+            mapclassify classifier.
+        preset_max : int
+            Maximum value to return when .max() is called. Typically this
+            should be set to the classifier ``k`` property, which is the number
+            of classes in the classifier.
+        preset_min : int, optional
+            Minimum value to return when .min() is called. Under most
+            circumstances, the default value (0) should be used.
+
+        Returns
+        -------
+        YBFixedBounds
+            New instance of YBFixedBounds with present min() and max() values.
+        """
+        array = np.asarray(input_array).view(cls)
+        array.__dict__.update(
+            {
+                "_preset_max": preset_max,
+                "_preset_min": preset_min,
+            }
+        )
+        return array
+
+    def max(self):
+        """Return preset maximum value."""
+        return self._preset_max
+
+    def min(self):
+        """Return preset minimum value."""
+        return self._preset_min
+
+
+# pylint: disable=dangerous-default-value,too-many-arguments,too-many-branches
+def map_geodataframe_column(
+    data_df,
+    column,
+    color_map="viridis",
+    breaks=None,
+    map_title=None,
+    legend_title=None,
+    background_df=None,
+    boundaries_df=None,
+    extent=None,
+    boundaries_kwargs={"linewidth": 0.75, "zorder": 1, "edgecolor": "white"},
+    layer_kwargs={},
+    legend_kwargs={"marker": "s", "frameon": False, "bbox_to_anchor": (1, 0.5),
+                   "loc": "center left"},
```

Comment:
Mike likes to live life on the edge! Is your thought here that users will never be able to access these dictionaries (and you of course will be very careful not to update them between calls of this function), so you are "safe" to use them as default parameters? I won't get in the way here, but it certainly puts a pit in my stomach